src/sidebar/services/thumbnail.ts

```diff
@@ -9,12 +9,38 @@ import type { FrameSyncService } from './frame-sync';
 export class ThumbnailService {
   /** Map of annotation tag to thumbnail image. */
   #cache = new Map<string, ImageBitmap>();
+
+  /** Max thumbnails to keep in cache. */
+  #cacheSize = 30;
+
   #frameSync: FrameSyncService;
 
   constructor(frameSync: FrameSyncService) {
     this.#frameSync = frameSync;
   }
 
+  /** Return the number of thumbnails that are cached. */
+  get cacheSize() {
+    return this.#cacheSize;
+  }
+
+  /** Set the number of thumbnails that are cached. */
+  set cacheSize(size: number) {
+    this.#cacheSize = size;
+    this.#pruneCache();
+  }
+
+  #pruneCache() {
+    // Keys are visited in insertion order, so the least recently used entry
+    // will be visited first.
+    for (const tag of this.#cache.keys()) {
+      this.#cache.delete(tag);
+      if (this.#cache.size <= this.cacheSize) {
+        break;
+      }
+    }
```

Comment:
I checked what happens if modifying a Map during iteration. In JS this is [safe to do](https://stackoverflow.com/a/76400380/434243). AFAIK that is not the case in Python.